src/app/page.tsx

```diff
@@ -1,39 +1,18 @@
-import { auth } from '@/actions/auth';
+import { getQueryClient } from '@/app/(provider)/get-query-client';
+import { dehydrate, HydrationBoundary } from '@tanstack/react-query';
+import Section from './section';
+import { employeesOptions } from '@/actions/query-options';
 
-const getEmployees = async () => {
-  try {
-    const res = await fetch('https://dummy.restapiexample.com/api/v1/employees');
-    const employees = await res.json();
+export default function Home() {
+  const queryClient = getQueryClient();
 
-    return employees.data;
-  } catch (error) {
-    console.error(error);
-  }
-};
-
-export default async function Home() {
-  const session = await auth();
-  const employees = await getEmployees();
+  void queryClient.prefetchQuery(employeesOptions);
```

Comment:
await가 아닌 void로 처리하신 이유가 궁금합니다.

Reply:
```tsx
export default function Home() {
  const queryClient = getQueryClient();

  void queryClient.prefetchQuery(employeesOptions);

  return (
    <HydrationBoundary state={dehydrate(queryClient)}>
      <main>
        <Section />
      </main>
    </HydrationBoundary>
  );
}
```

수정된 코드를 보시면 async 함수를 사용하지 않았습니다.
따라서 함수 내부에서는 await 을 쓰지 못합니다.

> await' expressions are only allowed within async functions and at the top levels of modules.ts

다른 언어에서는 void 를 반환 없을때 일반적으로 사용하는데 보통 일반적으로 컴포넌트 내부에서 바로 함수를 호출한다고 하면 의아할 수도 있어서 명시적으로 보이기 위해서 넣어줬는데 찾아보니 잘 사용은 안하네요. 빼도 될거 같습니다.

왜 저게 가능한지에 대해 궁금하신거 같아 간단하게 설명드리자면
원래는 기존 prefetch 는 await 으로 중단했어야 했습니다. 하지만 react query 5.40 버전부터 pending 상태의 쿼리도 hydration 가능하게 되면서 await 을 안해도 됩니다. 문서를 보시면 기존의 prefetch 는 success status query 만 hydration 가능했는데 지금은 데이터가 없는 준비 상태 즉, 대기중인 상태의 쿼리들을 미리 next streaming 기능으로 내려 보낼 수 있게 됩니다.

> 사용시 주의할점
Note that you could also useQuery instead of useSuspenseQuery, and the Promise would still be picked up correctly. However, NextJs won't suspend in that case and the component will render in the pending status, which also opts out of server rendering the content.

useQuery 를 사용하면 해당 컴포넌트가 서버 렌더링이 진행되지 않은 상태로 옵니다. 그래서 따로 loading 처리를 해줘야 합니다.

[링크로 이동](https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr#streaming-with-server-components)해서 문서를 읽어보시면 자세한 이유가 나옵니다.

Reply:
아하 감사합니다 !!